app/scripts/modules/core/src/api/network.interceptor.ts

```diff
@@ -0,0 +1,58 @@
+import {
+  IDeferred, IHttpInterceptor, IHttpPromiseCallbackArg, IHttpProvider, IPromise, IQService, IWindowService, module
+} from 'angular';
+
+/**
+ * Handles two scenarios:
+ *   1. computer loses network connection (retries connections when network returns)
+ *   2. requests are aborted due to a network change (retries immediately)
+ */
+
+export class NetworkInterceptor implements IHttpInterceptor {
+
+  private networkAvailable: IDeferred<void>;
+
+  constructor(private $q: IQService,
+              private $window: IWindowService,
+              private $injector: any) {
```

Comment:
I don't know why, but things were blowing up when I tried to import this via `import IInjectorService = angular.auto.IInjectorService;`. It was also forcing me to cast an unbelievable amount of stuff in the `responseError` method, so I took the path of least resistance and left it as an `any`.